### PR TITLE
Segment Updates: BD-74, BD-171, BD-268, BD-314, BD-353

### LINKED ISSUE #640: Segment -> Braze Integration: Custom REST API should read REST API Instance
# Submit an Issue
**Describe the Problem:**
Currently the documentation cites that the Custom REST API Endpoint" does not exist as Braze does not have custom endpoints for REST API calls. After validating with SA + Product it appears that this field should actually correspond to the client's REST API endpoint for their instance (e.g. rest.iad-03.braze.com).

**Related Links, Resources, Code Samples, and Images:**
https://www.braze.com/docs/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment/#understand-the-difference-between-customer-api-endpoint-vs-custom-rest-api-endpoint

### Is this change associated with a Braze feature/product release?
No

## Before You Submit

- [ ] Fill out the relevant items under __Submit an Issue__ above.
- [ ] [Label](https://github.com/Appboy/braze-docs/issues/labels) this issue.
- [ ] Assign this issue to either yourself, the Technical Writer (@EmilyNecciai), or to another appropriate person. _If you do not assign anyone to an issue, your issue may be closed or left uncompleted._

---
---

---
---

[Docs Request Policy (Braze Internal Only)](https://confluence.braze.com/display/GROW/Request+Process)

### PR DESCRIPTION
**Pull request by EmilyNecciai**
# To Do

- [ ] Server to server screenshot

- [ ] general Revision

- [ ] Add Personas Info

Closes #640

### CODE REVIEW

**Comment by EmilyNecciai** [comment]
@dwillislc Would love your input on this!

I still have a few more links to add in this doc, but I would love to write the iOS, Android, and Web integration docs more thoroughly for the side-by-side integration.

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 1]
```diff
@@ -3,21 +3,25 @@ nav_title: Segment
...
 
 If you're looking for information on the Currents integration with Segment, [click here]({{ site.baseurl }}/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_for_currents/).
 
-# Pre-Requisites
+## Pre-Requisites
 
 | Requirement| Origin| Access| Description|
 | ---| ---| ---|
 | Segment Account & Account Information | Segment | https://app.segment.com/login | You must have an active Segment Account to utilize their services with Braze. |
+| Installed Source and Segment Source Libraries | Segment | https://segment.com/docs/sources/ | The origin of any data sent into Segment, such as mobile apps, websites, or backend servers. <br> <br> You must install the the libraries into your app, site, or server before being able to set up a successful `Source -> Destination` flow.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: In the sentence "You must install the the", there's an extra `the`.

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 2]
```diff
@@ -48,9 +52,9 @@ When configuring Braze from Segment, you'll have many options to customize the f
 |Track Only Named Pages | On/Off (True/False) | Sends all [named Segment page calls](https://segment.com/docs/spec/page/) to Braze
 |Update Existing Users Only| On/Off (True/False) | This only applies to Server Side integrations. This determines whether or not all users vs. existing users will be updated. This defaults to false. |
 
-# Side-by-Side SDK Integration
+## Side-by-Side SDK Integration
 
-Braze's side-by-side integration maps Segment's SDK to ours, allowing access to deeper features and a more comprehensive usage of Braze than the server-to-server integration. These mappings of Segment's SDK for [Android][31], [iOS][32] and [Analytics.js][33] (Segment's Javascript SDK) are open source and can be found on our GitHub page.
+Also called "Cloud Mode", a side-by-side integration maps Segment's SDK to Braze's, allowing access to deeper features and a more comprehensive usage of Braze than the server-to-server integration. These mappings of Segment's SDK for [Android][31], [iOS][32] and [Analytics.js (Segment's Javascript SDK)][33]  are open source and can be found on our GitHub page.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: SBS is "Device Mode" not "Cloud Mode".

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 3]
```diff
@@ -66,43 +70,43 @@ The keys required to input in Segment's dashboard can be found in the [Developer
 
 For more details regarding Braze's SDKs, please refer to our [iOS][34], [Android][35] and [Web][38] documentation.
 
-# Server-to-Server Integration {#segment-1}
+## Server-to-Server Integration {#segment-1}
 
-The server-to-server integration forwards data from Segment to Braze's REST API. This integration is **only** used in association with Segment's [server-side libraries][36], such as their Ruby or Go SDKs. Enable the integration by setting your [App Group's REST API Key][39] and Braze's [REST API endpoint][40] for your corresponding data center in your destination settings on Segment's dashboard.
+Also called "Device Mode", the server-to-server integration forwards data from Segment to Braze's REST API. This integration is **only** used in association with Segment's [server-side libraries][36], such as their Ruby or Go SDKs. Enable the integration by setting your [App Group's REST API Key][39] and Braze's [REST API endpoint][40] for your corresponding data center in your destination settings on Segment's dashboard.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: S2S is "Cloud Mode"

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## About Segment Personas
+
+[Personas](https://segment.com/docs/personas/) is Segment's built-in audience builder. You can go to your Segment Dashboard and create segments of users based on data you have already collected across various sources.
+
+From there, these segments will be assigned a [custom attribute]({{ site.baseurl }}/user_guide/data_and_analytics/custom_data/custom_attributes/) in a Braze user profile (only for users who already have user IDs) that designates whether they fall into a certain audience or not.
+
+## Requirements
+
+Before can access and use Segment Personas, you must have already [set up Braze as a destination]({{ site.baseurl }}/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment/) your Segment integration, including inputting the correct "Appboy Data Center" and "Braze REST API Key" into your destination [Connection Settings]({{ site.baseurl }}/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment/#connection-settings).
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md]: "Before can access and use..." should be "Before you can access and use"

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Sync Time
+
+Though the default setting for the Braze to Segment Personas connection is `Realtime`, there are some filters which will disqualify the persona from syncing in real time, including some time-based filters which restrict your audience's size at the time of message send.
+
+## Segment Replays
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md]: This is a separate feature from Personas i believe. But it looks like it's being nested under Personas in our docs?

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+Segment provides a service to clients to "Replay" all historical data to a new technology partner. New Braze customers who want to import all relevant historical data can do so through Segment.
+
+Segment will connect to our [users/track endpoint]({{ site.baseurl }}/api/endpoints/user_data/#user-track-endpoint) to import user data into Braze on behalf of the client.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md]: If users do not have an external ID, they will not be imported into braze as our users/track endpoint requires a user ID if a Braze ID or user alias is not provided. Currently, Segment does not map to Braze's Braze ID or user alias, so all anonymous data will not be replayed over.

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Segment Debugger Testing
+
+Segment's dashboard provides a "Debugger" feature that allows customers to test whether data from a "Source" is transferring to a "Destination" as expected.
+
+This feature connects to Braze's [users/track endpoint]({{ site.baseurl }}/api/endpoints/user_data/#user-track-endpoint), meaning that it can only be used for identified users (users who already have a user ID for their Braze user profile).
+
+Before can access and use Segment Personas, you must have already [set up Braze as a destination]({{ site.baseurl }}/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment/) your Segment integration, including inputting the correct "Appboy Data Center" and "Braze REST API Key" into your destination [Connection Settings]({{ site.baseurl }}/partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment/#connection-settings)
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment_personas.md]: "Before _you_ can"

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 8]
```diff
@@ -81,26 +81,26 @@ You can learn more about Segment's integration options (Connection Modes), inclu
 
 ### Side-by-Side SDK Integration
 
-Also called "Cloud-mode", this integration maps Segment's SDK to Braze's, allowing access to deeper features and a more comprehensive usage of Braze than the server-to-server integration.
+Also called "Cloud-mode", this integration maps Segment's SDK and [methods](#methods) to Braze's, allowing access to deeper features and a more comprehensive usage of Braze than the server-to-server integration.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: called "Device Mode"

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 9]
```diff
@@ -111,94 +107,138 @@ For Braze's Web SDK, [Segment's Analytics.js library][33] dynamically pulls in a
 For the side-by-side integration, inputting the "Braze REST API Key" into your [Braze's Connection Settings](#connection-settings) in the Segment Dashboard is **not** necessary.
 {% endalert %}
 
-### Methods
+### Server-to-Server Integration
 
-Braze supports the [Identify](https://segment.com/docs/spec/identify/), [Track](https://segment.com/docs/spec/track/), and [Page](https://segment.com/docs/spec/page/) (for mobile) / [Screen](https://segment.com/docs/spec/screen/) (web) Segment methods; however, our REST APIs require you to include a [user ID][41] when making these calls.
+Also called "Device-mode", this integration forwards data from Segment to Braze's REST API.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: "Cloud Mode"

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 10]
```diff
@@ -111,94 +107,138 @@ For Braze's Web SDK, [Segment's Analytics.js library][33] dynamically pulls in a
...
+Some [automatically captured][25] data is only available through the side-by-side integration. The following data is __not available via the server-to-server integration__:
+- Sessions
+- First Used App
+- Last Used App
 
-##### Completed Order
+#### Enabling Push Notifications
 
-When you _track_ an event with the name `Completed Order` using the format described in Segment's [ECommerce API][4], we will record the products you've listed as [purchases][12].
+Currently, Braze's server-to-server integration with Segment __does not__ support methods for push tokens. In order to enable push notifications in Braze, you must import push tokens via the [User Attribute Object][18] of our [User Data][19] REST API.
```
Comment (dwillislc) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: Or rely on the SBS integration for push token capture and mapping

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 11]
```diff
@@ -46,99 +59,198 @@ When configuring Braze from Segment, you'll have many options to customize the f
...
+| [Track with Properties](https://segment.com/docs/spec/track/) | Logged as [Event Property]({{ site.baseurl }}/user_guide/data_and_analytics/custom_data/custom_events/#custom-event-properties). | `analytics.track("played_game", {name: "BotW", weapon: "boomerang"});` > `appboy.logCustomEvent("played_game", { "name": "BotW", "weapon": "boomerang"});` |
+| [Track with Product](https://segment.com/docs/spec/track/) | Logged as a [Purchase Event]({{ site.baseurl }}/developer_guide/platform_integration_guides/web/analytics/logging_purchases/). | `analytics.track("purchase", {products: [product_id: "ab12", price: 19]});` > `appboy.logPurchase("ab12", 19);` |
 
-Currently, Braze's server-to-server integration with Segment does not support methods for push tokens. In order to enable push notifications in Braze, you must import push tokens via the [User Attribute Object][18] of our [User Data][19] REST API.
+#### Completed Order
 
-## How To Tell If You're Receiving Data From Segment
+When you _track_ an event with the name `Completed Order` using the format described in Segment's [ECommerce API][4], we will record the products you've listed as [purchases][12].
```
Comment (KellieHawks) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: Link 12 directs to 404 page

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 12]
```diff
@@ -46,99 +59,198 @@ When configuring Braze from Segment, you'll have many options to customize the f
...
+|---|---|---|
+| [Page](https://segment.com/docs/spec/page/)/[Screen](https://segment.com/docs/spec/screen/) without name	| Logged as a [Custom Event][13] |	`analytics.page();` > 	`appboy.logCustomEvent("Loaded a Page");` |
+| [Page](https://segment.com/docs/spec/page/)/[Screen](https://segment.com/docs/spec/screen/) with name |	Logged as a [Custom Event][13]	| `analytics.page("Home");`	> `appboy.logCustomEvent("Viewed Home Page");` |
 
-Consequently, most of your [app usage dashboard][27] (lifetime sessions, MAU, DAU, Stickiness, Daily Sessions and Daily Sessions per MAU) will be blank even if Braze is receiving data from Segment.
+## Step 3: Test Your Integration
+
+Most of your [app usage dashboard][27] (lifetime sessions, MAU, DAU, Stickiness, Daily Sessions and Daily Sessions per MAU) will be blank even if Braze is receiving data from Segment.
 
 You can view your data in the [custom event dashboard][22], the [revenue dashboard][28] or by [creating a segment][23]. The custom event dashboard allows you to view custom event counts over time. Note that you will not be able to use [formulas][24] that include MAU and DAU statistics.
```
Comment (KellieHawks) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: link 22 points to 404 link

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md [hunk 13]
```diff
@@ -46,99 +59,198 @@ When configuring Braze from Segment, you'll have many options to customize the f
...
 
-If you're sending purchase data to Braze (see [Completed Order][4]), the [revenue dashboard][28] allows you to view data on revenue or purchases over specific periods of time or your app's total revenue.
+If you're sending purchase data to Braze (see [Completed Order][1]), the [revenue dashboard][28] allows you to view data on revenue or purchases over specific periods of time or your app's total revenue.
```
Comment (KellieHawks) [inline _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: link 1 points to broken link

File: _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md (hunk 11 above)
Comment (EmilyNecciai) [reply _docs/_partners/technology_partners/data_and_infrastructure_agility/customer_data_platform/segment.md]: Agh. Thank you!

**Pull request merged**